src/index.ts

```diff
@@ -207,25 +212,73 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
       throwOnFailedInitialization: true, // always use true here as underlying instance fetch is surrounded by try/catch
       skipInitialPoll: config.skipInitialRequest ?? false,
     };
+    EppoJSClient.instance.setConfigurationRequestParameters(requestConfiguration);
 
-    EppoJSClient.instance.setLogger(config.assignmentLogger);
+    // We have two at-bats for initialization: from the configuration store and from fetching
+    // We can resolve the initialization promise as soon as either one succeeds
+    let initFromConfigStoreError = undefined;
+    let initFromFetchError = undefined;
 
-    // default behavior is to use a LocalStorage-based assignment cache.
-    // this can be overridden after initialization.
-    EppoJSClient.instance.useCustomAssignmentCache(new LocalStorageAssignmentCache());
-    EppoJSClient.instance.setConfigurationRequestParameters(requestConfiguration);
-    await EppoJSClient.instance.fetchFlagConfigurations();
+    const attemptInitFromConfigStore = configurationStore
+      .init()
+      .then(() => {
+        if (!configurationStore.getKeys().length) {
+          // Consider empty configuration stores invalid
+          console.warn('Empty cached configuration');
+          initFromConfigStoreError = new Error('Configuration store was empty');
+          return '';
+        }
+        return 'config store';
+      })
+      .catch((e) => {
+        console.warn('Eppo SDK encountered an error initializing from the configuration store', e);
+        initFromConfigStoreError = e;
+      });
+    const attemptInitFromFetch = EppoJSClient.instance
+      .fetchFlagConfigurations()
```

Comment:
Does `skipInitialPoll` continue to work as before - I assume if it is `true` then `fetchFlagConfigurations` returns.

Reply:
Yeah if that is true then `fetchClagConfigurations()`'s call of `await this.requestPoller.start();` will complete without the initial poll attempt firing off.